daffodil-runtime1-unparser/src/main/scala/edu/illinois/ncsa/daffodil/processors/unparsers/PackedBinaryUnparserTraits.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.illinois.ncsa.daffodil.processors.unparsers
+
+import edu.illinois.ncsa.daffodil.processors.ElementRuntimeData
+import edu.illinois.ncsa.daffodil.processors.ParseOrUnparseState
+import edu.illinois.ncsa.daffodil.util.Maybe._
+import java.lang.{ Number => JNumber }
+import java.math.{ BigDecimal => JBigDecimal }
+import edu.illinois.ncsa.daffodil.exceptions.Assert
+import edu.illinois.ncsa.daffodil.io.DataOutputStream
+import edu.illinois.ncsa.daffodil.io.FormatInfo
+import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.BinaryNumberCheckPolicy
+
+trait PackedBinaryConversion {
+  def fromBigInteger(num: String, nBits: Int, signCodes: String, policy: BinaryNumberCheckPolicy): Array[Byte]
+}
+
+abstract class PackedBinaryBaseUnparser(
+  e: ElementRuntimeData,
+  binaryPackedSignCodes: String = "",
+  binaryNumberCheckPolicy: BinaryNumberCheckPolicy = BinaryNumberCheckPolicy.Strict)
+  extends PrimUnparserObject(e)
+  with PackedBinaryConversion {
+
+  protected def getBitLength(s: ParseOrUnparseState): Int
+
+  def putNumber(dos: DataOutputStream, number: JNumber, nBits: Int, finfo: FormatInfo): Boolean = {
+
+    val num: String = number.isInstanceOf[JBigDecimal] match {
+      case true => number.asInstanceOf[JBigDecimal].toPlainString
+      case false => number.toString
+    }
+
+    dos.putByteArray(fromBigInteger(num, nBits, binaryPackedSignCodes, binaryNumberCheckPolicy), nBits, finfo)
```

Comment:
Can you split this up into multiple lines? It's a bit hard to read and see which parameters go to which function.